static/js/modules/typeahead.js

```diff
@@ -52,8 +53,13 @@ module.exports = {
         url = "/rest/names?q=%QUERY";
 
     if (typeof API_LOCATION !== 'undefined') {
-        url = "/names?q=%QUERY&api_key=6wpqET5GjxCwAyBqXjc4dWrfMdshFE6wu3txb6yZ"; // Replace with the correct api key
-        url = API_LOCATION.concat(url);
+        url = URI(API_LOCATION)
+          .path([API_VERSION, 'names'].join('/'))
+          .query({
+            q: '%QUERY',
+            api_key: '6wpqET5GjxCwAyBqXjc4dWrfMdshFE6wu3txb6yZ'
```

Comment:
Yes. This was a quick hack to get things going for the demo this past
Tuesday.

On Thu, Apr 30, 2015 at 1:11 PM, Marco Segreto <notifications@github.com>
wrote:

> In static/js/modules/typeahead.js
> <https://github.com/18F/openFEC-web-app/pull/108#discussion_r29448969>:
>
> > @@ -52,8 +53,13 @@ module.exports = {
> >          url = "/rest/names?q=%QUERY";
> >
> >      if (typeof API_LOCATION !== 'undefined') {
> > -        url = "/names?q=%QUERY&api_key=6wpqET5GjxCwAyBqXjc4dWrfMdshFE6wu3txb6yZ"; // Replace with the correct api key
> > -        url = API_LOCATION.concat(url);
> > +        url = URI(API_LOCATION)
> > +          .path([API_VERSION, 'names'].join('/'))
> > +          .query({
> > +            q: '%QUERY',
> > +            api_key: '6wpqET5GjxCwAyBqXjc4dWrfMdshFE6wu3txb6yZ'
>
> is api_key supposed to be a constant here? Shouldn't it be reading it from
> config somehow?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/18F/openFEC-web-app/pull/108/files#r29448969>.
>



-- 
Alison Rowland
Technical Lead, https://fbopen.gsa.gov | 18F
202-317-0124